query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -41,7 +41,12 @@ class MinOverTimeChunkedFunctionD(var min: Double = Double.MaxValue) extends Chu
     val it = doubleReader.iterate(doubleVect, startRowNum)
     while (rowNum <= endRowNum) {
       val nextVal = it.next
-      if (!JLDouble.isNaN(nextVal)) min = Math.min(min, nextVal)  // cannot compare NaN, always < anything else
+      if (!JLDouble.isNaN(nextVal)) {
+        if (min.isNaN) {
+          min = Double.MaxValue
```

Comment:
Slightly faster: instead of setting Double.MaxValue just set it to the nextVal because that becomes the max.  IE:

    min = if (min.isNan) nextval else Math.min(min, nextval)

Reply:
Good suggestion! Made the change